### PR TITLE
Migrate DefaultConsumption to SQL (2 of 3)

### PR DESCRIPTION
**Pull request by orangejenny**
Branches off of https://github.com/dimagi/commcare-hq/pull/26602, first new commit is the one marked "PR2".

[QA ticket](https://dimagi-dev.atlassian.net/browse/QA-1079)

### CODE REVIEW

File: corehq/apps/consumption/shortcuts.py [hunk 1]
```diff
@@ -17,39 +15,32 @@ def get_default_monthly_consumption(domain, product_id, location_type, case_id):
     parameters.
     """
 
-    keys = [
```
Comment (orangejenny) [inline corehq/apps/consumption/shortcuts.py]: @dannyroberts This function is the couch question I had for you. My understanding of it is that using an array of arrays of keys returns all documents that match the first key, then all that match the second, and so on.

**Comment by orangejenny** [comment]
Rebased to clean up commit history.

**Review by kaapstorm — APPROVED** [review]

**Review by millerdev — APPROVED** [review]

File: corehq/apps/consumption/shortcuts.py [hunk 2]
```diff
@@ -97,12 +97,12 @@ def build_consumption_dict(domain):
     SQLDefaultConsumption.objects.filter(domain=domain)
 
     return dict(
-        (tuple(
+        (tuple([
```
Comment (millerdev) [inline corehq/apps/consumption/shortcuts.py]: nit: could use a literal tuple
```suggestion
        ((
```

**Review by millerdev — APPROVED** [review]

File: corehq/apps/consumption/shortcuts.py [hunk 3]
```diff
@@ -85,30 +84,26 @@ def _update_or_create_default(domain, amount, default, type, **kwargs):
...
+    Takes raw rows from couch and builds a dict to
     look up consumption values from.
     """
-    raw_rows = get_domain_monthly_consumption_data(domain)
+    SQLDefaultConsumption.objects.filter(domain=domain)
 
     return dict(
-        (hashable_key(row['key']), Decimal(row['value']))
-        for row in raw_rows if row['value']
+        (tuple((
```
Comment (millerdev) [inline corehq/apps/consumption/shortcuts.py]: Wait, `tuple` is not necessary. Comma-separated items (often wrapped in parens) represent a tuple-literal.
```suggestion
        (((
```

Could also get rid a superfluous set of parens.

File: corehq/apps/consumption/shortcuts.py [hunk 4]
```diff
@@ -85,30 +84,26 @@ def _update_or_create_default(domain, amount, default, type, **kwargs):
...
+            obj.domain,
+            obj.product_id,
+            obj.supply_point_type,
+            obj.supply_point_id,
+        )), obj.default_consumption)
+        for obj in SQLDefaultConsumption.objects.filter(domain=domain) if obj.default_consumption
     )
```
Comment (kaapstorm) [inline corehq/apps/consumption/shortcuts.py]: At the risk of getting bogged down in details, I'll throw in my 5c: You could use a dictionary comprehension. And that `hashable_key()` function would make the comprehension more readable, so I'd advocate resurrecting it:

```suggestion
    return {
        hash_key(obj): obj.default_consumption
        for obj in SQLDefaultConsumption.objects.filter(domain=domain)
        if obj.default_consumption
    }
    
def hash_key(obj):
    return (
        obj.domain,
        obj.product_id,
        obj.supply_point_type,
        obj.supply_point_id,
    )
```

**Review by kaapstorm — APPROVED** [review]

**Comment by orangejenny** [comment]
@millerdev @kaapstorm Updated `build_consumption_dict` in 7d0a6101289

**Review by millerdev — APPROVED** [review]

**Review by kaapstorm — APPROVED** [review]

File: corehq/apps/consumption/shortcuts.py [hunk 5]
```diff
@@ -91,19 +91,23 @@ def _update_or_create_default(domain, amount, default, type, **kwargs):
 
 def build_consumption_dict(domain):
     """
-    Takes raw rows from couch and builds a dict to
-    look up consumption values from.
+    Builds a dict to look up consumption values from.
     """
     SQLDefaultConsumption.objects.filter(domain=domain)
```
Comment (kaapstorm) [inline corehq/apps/consumption/shortcuts.py]: I feel bad adding another comment :grimacing: but I don't think this line does anything. Am I missing something?
```suggestion
```

Comment (orangejenny) [reply corehq/apps/consumption/shortcuts.py]: huh, must have been a copy paste error. fixed.

**Comment by orangejenny** [comment]
Noting that this is on hold because PNA has a lot of these docs, so I'd like to handle the migration for them, but their deploy is currently blocked because they're on an old version of Ubuntu. So I'm waiting on this migration until either a) PNA upgrades Ubuntu or b) I improve the migration workflow so it'll be easier for them to migrate themselves (right now they'd have to check out an old commit and log into their machines to run the migration, without good commcare-cloud support).

@dannyroberts https://github.com/dimagi/commcare-hq/pull/26601 breaks deploy for old versions of linux because the [PPA we get node.js from ](https://github.com/dimagi/commcare-cloud/blob/master/src/commcare_cloud/ansible/roles/nodejs/tasks/trusty_apt_repos.yml) doesn't have a recent node version for ubuntu 14.04, which ultimately prevents you from installing less from npm at the correct version. Discussed with @AndreaMKing5 and determined this isn't a problem because we don't support ubuntu that old, just fyi.

**Comment by orangejenny** [comment]
Migration process was improved in https://github.com/dimagi/commcare-hq/pull/26758, and QA has passed. Merging.

**Pull request merged**